core/lib/tee_verifier/src/lib.rs

```diff
@@ -0,0 +1,732 @@
+    chain_id: L2ChainId,
+}
+
+impl From<V1SystemEnv> for SystemEnv {
+    fn from(value: V1SystemEnv) -> Self {
+        let V1SystemEnv {
+            zk_porter_available,
+            version,
+            base_system_smart_contracts,
+            bootloader_gas_limit,
+            execution_mode,
+            default_validation_computational_gas_limit,
+            chain_id,
+        } = value;
+
+        SystemEnv {
+            default_validation_computational_gas_limit,
+            chain_id,
+            zk_porter_available,
+            version,
+            execution_mode: execution_mode.into(),
+            base_system_smart_contracts: base_system_smart_contracts.into(),
+            bootloader_gas_limit,
+        }
+    }
+}
+
+impl From<SystemEnv> for V1SystemEnv {
+    fn from(value: SystemEnv) -> Self {
+        let SystemEnv {
+            zk_porter_available,
+            version,
+            base_system_smart_contracts,
+            bootloader_gas_limit,
+            execution_mode,
+            default_validation_computational_gas_limit,
+            chain_id,
+        } = value;
+
+        V1SystemEnv {
+            default_validation_computational_gas_limit,
+            chain_id,
+            zk_porter_available,
+            version,
+            execution_mode: execution_mode.into(),
+            base_system_smart_contracts: base_system_smart_contracts.into(),
+            bootloader_gas_limit,
+        }
+    }
+}
+
+/// Shadowed struct of [`TxExecutionMode`]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+enum V1TxExecutionMode {
+    VerifyExecute,
+    EstimateFee,
+    EthCall,
+}
+
+impl From<V1TxExecutionMode> for TxExecutionMode {
+    fn from(value: V1TxExecutionMode) -> Self {
+        match value {
+            V1TxExecutionMode::VerifyExecute => TxExecutionMode::VerifyExecute,
+            V1TxExecutionMode::EstimateFee => TxExecutionMode::EstimateFee,
+            V1TxExecutionMode::EthCall => TxExecutionMode::EthCall,
+        }
+    }
+}
+
+impl From<TxExecutionMode> for V1TxExecutionMode {
+    fn from(value: TxExecutionMode) -> Self {
+        match value {
+            TxExecutionMode::VerifyExecute => V1TxExecutionMode::VerifyExecute,
+            TxExecutionMode::EstimateFee => V1TxExecutionMode::EstimateFee,
+            TxExecutionMode::EthCall => V1TxExecutionMode::EthCall,
+        }
+    }
+}
+
+/// Version 1 of the data used as input for the TEE verifier.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct V1TeeVerifierInput {
+    prepare_basic_circuits_job: PrepareBasicCircuitsJob,
+    new_root_hash: H256,
+    l2_blocks_execution_data: Vec<V1L2BlockExecutionData>,
+    l1_batch_env: V1L1BatchEnv,
+    system_env: V1SystemEnv,
+    used_contracts: Vec<(H256, Vec<u8>)>,
+}
+
+/// Data used as input for the TEE verifier.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[non_exhaustive]
+pub enum TeeVerifierInput {
+    V0,
```

Comment:
Is this V0 thing needed?

Reply:
it avoids some rust warnings, which I would have to silence otherwise